library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/RegistrySynchronization.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.api.sync;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.SimpleRegistry;
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.qsl.registry.impl.sync.RegistryFlag;
+import org.quiltmc.qsl.registry.impl.sync.SynchronizedRegistry;
+
+/**
+ * Methods for manipulation of registry sync
+ */
+@ApiStatus.Experimental
+public final class RegistrySynchronization {
+
+	/**
+	 * Marks registry status as dirty, rebuilding its sync data
```

Comment:
The javadoc says that this method rebuilds sync data, but the implementation does no such thing!
```suggestion
	 * Marks registry status as dirty, requiring its synchronization data to be rebuilt.
```